source/pkg/reconciler/streamsource/controller.go

```diff
@@ -42,6 +44,34 @@ type envConfig struct {
 	Image string `envconfig:"STREAMSOURCE_RA_IMAGE" required:"true"`
 }
 
+func WithRedis(cw *reconcilersource.ConfigWatcher, cmw configmap.Watcher) {
+	cmName := ConfigMapName()
+	obs := updateFromRedisConfigMap
+	if dcmw, ok := cmw.(configmap.DefaultingWatcher); ok {
+		dcmw.WatchWithDefault(corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Name: cmName},
+			Data:       map[string]string{},
+		}, obs)
+
+	} else {
+		cmw.Watch(cmName, obs)
+	}
+}
+
+func updateFromRedisConfigMap(cfg *corev1.ConfigMap) {
```

Comment:
I don't think this function is complete yet. Looking into what more needs to be done....